src/main/java/data/Recipe.java

```diff
@@ -17,41 +17,69 @@
 import java.util.logging.*;
 import java.util.Iterator;
 
-/**
- * Stores a recipe's data.
- */
+/** Stores a recipe's data. */
 public class Recipe {
 
   private static final Logger LOGGER = Logger.getLogger(Logger.GLOBAL_LOGGER_NAME);
 
   private String name;
   private String description;
+  private List<String> tags;
+  private List<String> ingredients;
   private List<Step> steps;
   private List<SpinOff> spinOffs;
 
-  /**
-   * Copy constructor called when creating spin-offs.
-   */
+  /** Copy constructor called when creating spin-offs. */
   public Recipe(Recipe recipe) {
     this.name = recipe.name;
     this.description = recipe.description;
+    this.tags = recipe.tags;
+    this.ingredients = recipe.ingredients;
     this.steps = recipe.steps;
     this.spinOffs = recipe.spinOffs;
   }
   
-  /**
-   * Default constructor called when creating a new recipe.
-   */
-  public Recipe(String name, String description, List<Step> steps) {
+  /** Default constructor called when creating a new recipe. */
+  public Recipe(String name, String description, List<String> tags, List<String> ingredients, List<Step> steps) {
     this.name = name;
     this.description = description;
+    this.tags = tags;
+    this.ingredients = ingredients;
```

Comment:
When using the copy constructor, both the original and the newly constructed will point to the same lists b/c the list is not actually being copied. This is also true of the strings, but that's less of an issue b/c they are immutable.

I think this is probably not intended.

Reply:
Good catch, thank you!